test/test_query.cpp

```diff
@@ -2011,6 +2009,19 @@ TEST(Query_SubtableExpression)
     tv = q.find_all();
     CHECK_EQUAL(tv.size(), 1);
     CHECK_EQUAL(tv.get_source_ndx(0), 1);
+
+    q = table->column<SubTable>(1).list<String>().begins_with(StringData("Str"));
+    tv = q.find_all();
+    CHECK_EQUAL(tv.size(), 3);
+    q = table->column<SubTable>(1).list<String>().ends_with(StringData("_8"));
+    tv = q.find_all();
+    CHECK_EQUAL(tv.size(), 1);
+    CHECK_EQUAL(tv.get_source_ndx(0), 2);
+    q = table->column<SubTable>(1).list<String>().begins_with(table->column<String>(2), false);
```

Comment:
That syntax doesn't compile in the current design, but it shouldn't be to hard to extend. If it makes sense, can you also add it to this PR @jedelbo ?